lib/router/index.js

```diff
@@ -59,6 +61,22 @@ function throwIfNoRouter () {
 // Export the SingletonRouter and this is the public API.
 export default SingletonRouter
 
+export function withRouter (Comp) {
+  const compName = Comp.displayName || Comp.name || 'Component'
+
+  return class ComponentWithRouter extends Component {
+    static contextTypes = {
+      router: PropTypes.object
+    }
+
+    static displayName = `withRouter(${compName})`
+
+    render () {
+      return <Comp {...this.props} router={this.context.router} />
```

Comment:
Could be useful: https://github.com/acdlite/recompose/blob/master/src/packages/recompose/createEagerFactory.js#L5.